tests/Tests.FeatureManagement/FeatureManagementTest.cs

```diff
@@ -1332,6 +1332,8 @@ public async Task VariantBasedInjection()
 
     public class FeatureManagementTelemetryTest
     {
+        private readonly AutoResetEvent _waitHandle = new AutoResetEvent(false);
```

Comment:
I didn't understand the reason why we force the thread to wait until the `ActivityStopped` callback is done.

Each time `featureManagemr.IsEnabledAsync` method is called, a new activity "FeatureEvaluation" will be started and only one activity event "feature_flag" will be added to the activity. Any execution of the callback is bounded to a specific activity.

Reply:
The main reason is that we need the test code to wait until all the listener handlers successfully run. Otherwise it will kick off all of the activities and finish immediately without waiting for proof that the listener succeeded. 

The quickest option would be a Wait(13) at the end to ensure we triggered this 13 (# of tests) times. Or some other waiting that would break once all tests have run.

I ended up preferring this one by one approach. It makes it easier to debug, easier to add more tests, and is pretty clear what's going on at each step. Surprisingly, this test is just as fast as our other tests so I didn't bother trying to speed it up.

Reply:
> Otherwise it will kick off all of the activities and finish immediately.

This doesn't sound reasonable to me. Is there any documentation about how `ActivityStopped` callback works?

Reply:
I'm incorrect- I thought the listeners were handled async- or at least not immediately at the end of the Activity. Turns out it is a synchronous callback that does happen right away. 

But still- I think it's fairly valuable in these tests because nothing is actually blocked unless the test fails. It also ensures the ActivityListener.OnActivityStopped is executed exactly once per call. Without doing a check like this, a typo might mean we don't run the OnActivityStopped callback at all and the tests would still pass (because no Assert failed).

We could replace this with a bool flipping around like "wasCallbackCalled".

> Is there any documentation about how ActivityStopped callback works?

I don't see great docs about it besides a shallow API reference and some random articles.

Reply:
> a typo (like the wrong activity name) might mean we don't run the OnActivityStopped callback at all and the tests would still pass (because no Assert failed).

Define a few booleans (e.g. `bool ranTestOne = false;`) and make sure in the callback to set them to true, at the end of the test assert they were flipped to assert that the callbacks were executed.

Shouldn't be any reason to have the synchronization primitive.

Reply:
After the change, I noticed there was a small bug here where the ActivityListener would be triggered from other tests running async (and using the same features). Added a thread check to mitigate that.

Reply:
Parent activity id should be available for this.

Reply:
Updated to use parent id.